homeassistant/components/snips/__init__.py

```diff
@@ -90,12 +90,8 @@
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Activate Snips component."""
-    # Make sure MQTT is available and the entry is loaded
-    if not hass.config_entries.async_entries(
-        mqtt.DOMAIN
-    ) or not await hass.config_entries.async_wait_component(
-        hass.config_entries.async_entries(mqtt.DOMAIN)[0]
-    ):
+    # Make sure MQTT integration is enabled and the client is available
+    if not await mqtt.async_wait_for_mqtt_client(hass):
```

Comment:
Why do we need it for snips?

Reply:
It needs `async_subscribe`
https://github.com/home-assistant/core/blob/0bcda9fe9cc5840c9605bc5d86f8e455e2ce43b4/homeassistant/components/snips/__init__.py#L166

Reply:
https://github.com/home-assistant/core/issues/81431

Reply:
Not sure though, I was thinking snips should be able to rely on the dependencies. But to avoid regression the check was added.

Reply:
Using the feature here is inconsistent with the documentation of the feature and our current practice of using dependencies.

I think we should clarify why we need it here or remove it.

Reply:
I think its ok to remove. I can't see how this `"dependencies": ["mqtt"],` wouldn't cover this case since its coming from `async_setup`

Reply:
I suggest we do that after merging #91454

Reply:
Opened https://github.com/home-assistant/core/pull/91790 to remove the checks